### PR TITLE
Created deprecated folder

### PR DESCRIPTION
**Pull request by tomashapl**
Created `_deprecated` folder and refactored deprecated components

- Container
- FieldFeedback
- Input
- InputText
- InputTextarea
- Radio
- Checkbox
- Select
- SystemMessage
- Typography

### CODE REVIEW

**Comment by Fallion** [comment]
Please rebase this.

File: src/index.js [hunk 1]
```diff
@@ -3,22 +3,13 @@ import defaultTheme from "@kiwicom/orbit-design-tokens";
...
-export { default as Section } from "./Container/Section";
-export { default as Select } from "./Select";
-export { default as SystemMessage } from "./SystemMessage";
-export { default as FieldFeedback } from "./FieldFeedback";
 export { default as TextLink } from "./TextLink";
-export { default as Typography } from "./Typography";
-export { default as Radio } from "./Radio";
 export { default as ThemeProvider } from "./Theming/ThemeProvider";
-export { Icons, defaultTheme, withTheme };
+export { Icons, DeprecatedComponents, defaultTheme, withTheme };
```
Comment (jaroslav-kubicek) [inline src/index.js]: This is breaking change ⚠️ 

I mean - I'm totally ok to move these components to deprecated folder, but it's nonwelcomed for me as a user of orbit to change imports all over the app if I want to bump the version.

If I currently use `Typography`, the goal is to get rid of using it in my code over time and replace it with new components => There might be some time when I use both deprecated and new components in code

With this change, I would be stuck with `0.0.0-rc7` because I don't want to change imports for components I plan to remove anyway.

If we intend to hint users these components should not be used, we should put `console.warn` or `console.error` into them.

TL;DR: Don't change how components can be imported.

File: src/_deprecated/Checkbox/index.js [hunk 2]
```diff
@@ -4,6 +4,9 @@ import * as React from "react";
 
 import Typography from "../Typography";
 import Icon from "./Icon";
+import deprecationWarning from "../../../config/deprecationWarning";
+
+deprecationWarning("Checkbox is deprecated and will be removed in next major release");
```
Comment (jaroslav-kubicek) [inline src/_deprecated/Checkbox/index.js]: deprecations warnings need to be in `render` function, this results in console warnings no matter what specifically is imported, since it's executed when file is imported, which is exactly what is happening in index:

```
import { Typography } from '...';
```

Comment (tomashapl) [reply src/_deprecated/Checkbox/index.js]: Yes, but if you add it to render then it warns when every render

Comment (jaroslav-kubicek) [reply src/_deprecated/Checkbox/index.js]: yes, IMHO this is more desired behaviour, early adoptions should be rewarded

No usage of deprecated components => no warning
Many usages of deprecated components => many warnings

Comment (tomashapl) [reply src/_deprecated/Checkbox/index.js]: Imagine `InputText` component, it rerenders everytime when you type into => sooo many warings, but one usage

also in index there isn't importing, it warns due to `*.stories.js` files

Comment (jaroslav-kubicek) [reply src/_deprecated/Checkbox/index.js]: > also in index there isn't importing, it warns due to *.stories.js files 

https://github.com/kiwicom/orbit-components/pull/174/files#diff-1fdf421c05c1140f6d71444ea2b27638R7

Comment (jaroslav-kubicek) [reply src/_deprecated/Checkbox/index.js]: It would be batched together in such case

Comment (tomashapl) [reply src/_deprecated/Checkbox/index.js]: OK, refactored, according to this i had to create new file inside components folder that includes pure component with export without HOC due to tests, because i couldn't get into component via `shallow` ¯\_(ツ)_/¯

Also left `index.js` file with HOC export, so it shouldn't break anything

File: config/DeprecatedComponent.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+// @flow
+import React from "react";
+
+const DeprecatedComponent = (Component: React$ComponentType<any>, text: string) => {
+  let warned = false;
+  return (props: Object) => {
```
Comment (jaroslav-kubicek) [inline config/DeprecatedComponent.js]: cool, just for future, it's possible to parametrize the types, I think something like:

```
const DeprecatedComponent = <Props>(Component: React$ComponentType<any>, text: string) => {
```

and then inside the function like `return (props: Props) => {`

Comment (tomashapl) [reply config/DeprecatedComponent.js]: Ok, thanks 😊
